src/csharp/Microsoft.Spark/Sql/Avro/Functions.cs

```diff
@@ -0,0 +1,94 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.Spark.Interop;
+using Microsoft.Spark.Interop.Ipc;
+
+namespace Microsoft.Spark.Sql.Avro
+{
+    /// <summary>
+    /// Functions for serialization and deserialization of data in Avro format.
+    /// </summary>
+    public static class Functions
+    {
+        private static IJvmBridge Jvm { get; } = SparkEnvironment.JvmBridge;
+        private static readonly Lazy<string> s_functionsClassName =
+            new(() =>
+            {
+                Version sparkVersion = SparkEnvironment.SparkVersion;
+                return sparkVersion.Major switch
+                {
+                    2 => "org.apache.spark.sql.avro.package",
+                    3 => "org.apache.spark.sql.avro.functions",
+                    _ => throw new NotSupportedException($"Spark {sparkVersion} not supported.")
+                };
+            });
+
+        /// <summary>
+        /// Converts a binary column of avro format into its corresponding catalyst value. The specified
+        /// schema must match the read data, otherwise the behavior is undefined: it may fail or return
+        /// arbitrary result.
+        /// </summary>
+        /// <param name="data">The binary column.</param>
+        /// <param name="jsonFormatSchema">The avro schema in JSON string format.</param>
+        /// <returns>Column object</returns>
+        [Since(Versions.V2_4_0)]
+        public static Column FromAvro(Column data, string jsonFormatSchema) => new(
```

Comment:
Looks like [target typed new expressions](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-9.0/target-typed-new) is part of the of C# 9.0 features and is also introduced with .NET 5.0 and the latest versions of Visual Studio.

Should we start adopting these new features?  Don't think this will work for users who aren't using either of the above , ie using the latest visual studio (maybe they are just using vs code), or if they don't have .NET core 5.0 installed.

Reply:
Hmm yeah makes sense, reverted.